.github/workflows/release.yml

```diff
@@ -260,15 +327,15 @@ jobs:
         working-directory: auto
         id: test_execution
         run: |
-          echo "## API tests ${{ matrix.db }} ${{ matrix.conf }}" >> $GITHUB_STEP_SUMMARY
+          echo "### API tests ${{ matrix.db }} ${{ matrix.conf }}" >> $GITHUB_STEP_SUMMARY
           if docker run --rm --network auto_default --env-file pytest.env -v ${{ github.workspace }}/reports:/app/reports \
-             ${{ steps.ecr.outputs.registry }}/tyk-automated-tests:${{ startsWith(github.ref_name, 'release-') && github.ref_name || 'master' }} \
-             pytest -c pytest_ci.ini --ci -m "not local and not dind ${{ matrix.markers }}" > tests.out; then
+             ${{ steps.ecr.outputs.registry }}/tyk-automated-tests:${{ needs.test-controller.outputs.gd_tag }} \
```

Comment:
@alephnull I'm afraid this won't work when merging against release code freeze branches like release-4.2 since there's no release-4.2 for tyk-automated-tests, it will just work for LTS branches since that's the only case where release lifecycled is shared between gw, dash and TAT

Reply:
Yes, it will not work for `release-4.2`. https://github.com/TykTechnologies/gromit/pull/242/commits/cc01cc1339270775a19223a9709bbb517457e59f ensures that this never runs for branches like `release-4.2`. AFAIK, these branches are temporary and have no long term need as each non-LTS release supersedes the previous one.

Reply:
Thinking about it further, I have reverted https://github.com/TykTechnologies/gromit/pull/242/commits/cc01cc1339270775a19223a9709bbb517457e59f as this change would prevent the workflow from running on code-freeze branches.
So to address branches like release-4.2, we could address it by tagging a release-4.2 for tyk-automated-tests when the branch is created. I think this is a bad process to follow and the bad practice of creating branches like `release-4.2` which have no value (afaict) should be discouraged.

Reply:
Pushed an implementation based on on regexps. LTS policy says that any minor could be an LTS but at present only minor version 0 is an LTS. If another minor version is declared LTS, then we'll have to use something like `(0|3)` which is not perfect but is good enough, I think.